src/daemon/rpmostreed-transaction-types.cxx

```diff
@@ -1148,11 +1168,30 @@ deploy_transaction_execute (RpmostreedTransaction *transaction,
       if (!rpmostree_origin_remove_all_packages (origin, &remove_changed, error))
         return FALSE;
     }
-  else if (self->uninstall_pkgs)
+  else
     {
-      if (!rpmostree_origin_remove_packages (origin, self->uninstall_pkgs,
-                                             idempotent_layering, &remove_changed, error))
-        return FALSE;
+      gboolean local_changed = FALSE;
+      if (self->uninstall_pkgs)
+        {
+          if (!rpmostree_origin_remove_packages (origin, self->uninstall_pkgs,
+                                                 idempotent_layering, &local_changed, error))
+            return FALSE;
+        }
+      remove_changed = remove_changed || local_changed;
```

Comment:
(Idle thought, we should change the origin API to do instead "keep value as TRUE if it was already", then this footgun goes away)